lib/index.d.ts

```diff
@@ -485,4 +487,29 @@ declare module 'openid-client' {
 
     [key: string]: unknown
   }
+
+  export class Strategy<TUser> {
+    constructor(options: IStrategyOptions, verify: StrategyVerifyCallbackUserInfo<TUser> | StrategyVerifyCallback<TUser>)
+  }
+
+  export interface IStrategyOptions {
+    client: IClient
+    /**
+     * Authorization Request parameters. The strategy will use these.
+     */
+    params: object
+    /**
+     * Boolean specifying whether the verify function should get the request object as first argument instead.
+     * Default: 'false'
+     */
+    passReqToCallback: boolean
+    /**
+     * The PKCE method to use. When 'true' it will resolve based on the issuer metadata, when 'false' no PKCE will be
+     * used. Default: 'false'
+     */
+    usePKCE: boolean | string
+  }
+
+  export type StrategyVerifyCallbackUserInfo<TUser> = (tokenset: TokenSet, userinfo?: object, done?: (err: any, user?: TUser) => void) => void)
```

Comment:
I had to make `done` optional because typescript will not allow an optional parameter followed by a required one.

Reply:
If I read [it correctly](https://github.com/panva/node-openid-client/blob/master/docs/README.md#new-strategyoptions-verify) (I'm not familiar with it), then neither `done` nor `userinfo` should be optional in this type. For use case without `userinfo` you have the type `StrategyVerifyCallback` below. Or do I'm missing something?

Reply:
`done` should not be optional, but because `userinfo` is, doing the following is illegal in TS:

`fn (userinfo?, done)`

I looked at the tests for the `Strategy` class in this project, and saw two usages for the verify callback:

`(tokenset, done)
`(tokenset, userinfo, done)`

The docs has `userinfo` as optional, and the tests for the `Strategy` also reflects both usages as well.

Reply:
But for this reason you have:
```ts
verify: StrategyVerifyCallbackUserInfo<TUser> | StrategyVerifyCallback<TUser>
```

Then it should be enough:
```ts
export type StrategyVerifyCallbackUserInfo<TUser> = (tokenset: TokenSet, userinfo: object, done: (err: any, user?: TUser) => void) => void
export type StrategyVerifyCallback<TUser> = (tokenset: TokenSet, done: (err: any, user?: TUser) => void) => void
```

Reply:
Oh, I see. Thanks! Will update.